server/graphql/types.js

```diff
@@ -45,10 +45,11 @@ const getCustomTypes = (strapi, nexus) => {
 					args: {
 						modelName: nexus.stringArg('The model name of the content type'),
 						slug: nexus.stringArg('The slug to query for'),
+						publicationState: nexus.stringArg('The publication state of the entry')
```

Comment:
Thanks for the review!

How would you go about checking whether a type has draftAndPublish enabled in the args?

I'd set up something like:
`...(checkForDraftAndPublish && {publicationState: nexus.stringArg('The publication state of the entry')})`
but I fail to wrap my head around how we can checkForDraftAndPublish prior to the resolve, so before we get any actual data from the model?

Other option I'm seeing would be to through a ValidationError in the resolve, if the param was given to a request for a type that doesn't have draftAndPublish enabled.

Thanks in advance!

Reply:
Sorry I made a comment but it looks to have been removed somehow?

Anyways, I would prefer the first option, we could break it out instead a function to build the args, but the problem I do not see a way to add args dependant on which type is being used in the union type. If you by any chance have any suggestions regarding this, let me know.

If not lets go with the second option of the validation error.

Reply:
No worries, thanks for getting back!

I did some more digging in the Nexus docs today and I think it won't be possible to add the publicationState arg based on the type of the union type of the entity that is being returned, since we can't really collect any info on the type of the entity that is being requested before we get the actual data returned (at least I don't see a way). 

If it's okay with you I'd write up the solution with throwing a validation error, even though it feels like a less elegant way...

Reply:
>I did some more digging in the Nexus docs today and I think it won't be possible to add the publicationState arg based on the type of the union type of the entity that is being returned, since we can't really collect any info on the type of the entity that is being requested before we get the actual data returned (at least I don't see a way).

Yeah I figured that was the case. I believe it is the reason I did not add this in the first place, same for filter params as well.

>If it's okay with you I'd write up the solution with throwing a validation error, even though it feels like a less elegant way...

Yeah less  elegant but seems to be the best alternative. Its fine with me, I am ok with using this method. I would recommend first updating from master as the file has changed quite a bit due to #33